### PR TITLE
Add equality operator to NuGetFramework

### PR DESCRIPTION
**Pull request by pranavkm**
This brings it at par with FrameworkName.

### CODE REVIEW

**Comment by pranavkm** [comment]
Ran into this with one of the tools in AspNet. Parsing some `NuGetFramework` values returns the same instances (because of the `FrameworkConstants` cache). Consequently `NuGetFramework.Parse("netcoreapp1.0") == NuGetFramework.Parse("netcoreapp1.0")`, but `NuGetFramework.Parse("netcoreapp1.1") != NuGetFramework.Parse("netcoreapp1.1")`. This fixes this oddity and brings it at par with `FrameworkName` that overloads this operator - https://msdn.microsoft.com/en-us/library/system.runtime.versioning.frameworkname(v=vs.110).aspx#Anchor_5

File: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs [hunk 1]
```diff
@@ -353,6 +353,16 @@ public bool Equals(NuGetFramework other)
...
         }
 
+        public static bool operator ==(NuGetFramework left, NuGetFramework right)
+        {
+            return Comparer.Equals(left, right);
+        }
+
+        public static bool operator !=(NuGetFramework left, NuGetFramework right)
+        {
+            return !(left == right);
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs:363]: super-nit: wouldn't negating `Comparer.Equals` be more consistent with the other implementations and have one fewer method invocations?

Comment (pranavkm) [reply src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs:363]: I basically copied what `FrameworkName` does - https://referencesource.microsoft.com/#System/sys/system/runtime/versioning/FrameworkName.cs,218. I guess it makes the equality conditions more localized. I'm open to change this if you feel strongly about it.

File: test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkTests.cs [hunk 2]
```diff
@@ -71,6 +82,76 @@ public void NuGetFramework_Equality(string a, string b)
         }
 
         [Theory]
+        [MemberData(nameof(EqualsFrameworkData))]
+        public void EqualityOperator_ReturnTrueForEqualsFramework(string a, string b)
+        {
+            // Arrange
+            var framework1 = NuGetFramework.Parse(a);
```
Comment (joelverhagen) [inline test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkTests.cs:89]: Please test `null` cases for both right and left sides.

Comment (joelverhagen) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkTests.cs:89]: And both `!=` and `==`.

File: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs:363]: Nope, seems fine.

**Comment by joelverhagen** [comment]
⌚️

**Comment by pranavkm** [comment]
🆙 📅

**Comment by joelverhagen** [comment]
👍 thanks for the tests. I'll work on running this through the CI. Is there any GitHub issue related to this change (on NuGet/Home or elsewhere)?

**Comment by pranavkm** [comment]
@joelverhagen, there isn't an issue for it. We ran in to this issue in one of our tools (https://github.com/aspnet/MvcPrecompilation/pull/31#discussion_r87677783) and the behavior was a bit surprising.

**Pull request merged**